### ISSUE TITLE
Running SCTransform before or after merging samples?

### ISSUE THREAD

**Issue by Victor21v** [satijalab/seurat#3046]
Hi,

I'm wondering whether I should merge two samples (Condition A and B, which I suppose they shouldn't have any or very few differences between them) before normalizing with SCTransform or not. In addition, I don't want to integrate the samples.

I'm not quite sure about the answer since if I normalize samples independently from each other and then merge them, I lose highly variable genes, although I can later set the variable genes with the intersect or union of the highly variable genes I got from the two samples, and then run PCA and UMAP. In addition, I don't know if I'm introducing batch effect normalizing samples independently from each other.

On the other hand, if I merge samples together, I suppose I have to regress out by the variable "Condition" to let the program know that I have two conditions and not just one to normalize.

So, to summarize, I don't know whether doing:

1) Run SCTransform on samples independently -> merge them -> Assign highly variable genes based on intersect or union.
2) Merge samples -> run SCTransform regressing out by variable "Condition" and downstream analysis.

Thanks in advance

**Comment by LeoCao-X** [comment]
Sorry to bother you, I have faced the same situation right now. But I haven't found any more relevant information. Do you have any new ideas? #2814 In this issue, @yuhanH suggested us to use all features in SCT to perform PCA. But I think the features number is too much and my not be the variable features we need.

**Comment by yuhanH** [comment]
Hi, @Victor21v 
One question. If you think condition A, and B have no batch effects, why do you want to regress out by variable "Condition"? 
What does your result look like if you run SCTransform on the merged object?

**Comment by yuhanH** [comment]
HI, @LeoCao-X 
Actually, in the https://github.com/satijalab/seurat/issues/2814 ,  I didn't suggest RunPCA by all features.  My point was that `FindVariableFeatures` function is designed for the log normalized data, so you cannot run it on the SCTransform data. 

However, after you run SCTransform with the default settings, the scale.data only contains the residues of variable genes. You can change the number of variables you need using the parameter, `variable.features.n` in the `SCTransform`.

**Comment by Rohit-Satyam** [comment]
@Victor21v @LeoCao-X Were you able to find out which approach is correct. I also have control and treatment sample that are very similar when I view them in UMAP. But I wish to understand if I should run SCT on control and treatment samples separately and then merge OR merge both samples first and then run SCT?
